### PR TITLE
Implement the underwater overlay property

### PR DESCRIPTION
**Pull request by maximumpower55**
Closes: #129

### CODE REVIEW

**Review by coderbot16 — CHANGES_REQUESTED** [review]
Would it be possible to control this with a method on WorldRenderingPipeline instead?

**Comment by coderbot16** [comment]
This is an example of how that might look:

https://github.com/IrisShaders/Iris/blob/528488f7c050d077edd76b0b7974aa14fb5ba9e2/src/main/java/net/coderbot/iris/mixin/MixinEntityRenderDispatcher.java

**Comment by maximumpower55** [comment]
> This is an example of how that might look:
> 
> https://github.com/IrisShaders/Iris/blob/528488f7c050d077edd76b0b7974aa14fb5ba9e2/src/main/java/net/coderbot/iris/mixin/MixinEntityRenderDispatcher.java

Will look into this tomorrow

File: src/main/java/net/coderbot/iris/mixin/MixinScreenEffectRenderer.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+@Mixin(ScreenEffectRenderer.class)
+public abstract class MixinScreenEffectRenderer {
+	@Inject(method = "renderWater", at = @At(value = "HEAD"), cancellable = true)
+	private static void iris$disableUnderWaterOverlayRendering(Minecraft minecraft, PoseStack poseStack, CallbackInfo ci) {
+		WorldRenderingPipeline pipeline = Iris.getPipelineManager().getPipelineNullable();
+
+		if (pipeline != null && !pipeline.shouldRenderUnderwaterOverlay()) {
+			ci.cancel();
+		}
+    }
```
Comment (coderbot16) [inline src/main/java/net/coderbot/iris/mixin/MixinScreenEffectRenderer.java:26]: wrong indentation?

**Review by coderbot16 — CHANGES_REQUESTED** [review]

File: src/main/java/net/coderbot/iris/mixin/MixinScreenEffectRenderer.java (hunk 1 above)
Comment (maximumpower55) [reply src/main/java/net/coderbot/iris/mixin/MixinScreenEffectRenderer.java:26]: Code_VyMWmCeXyD

i think thats just git

Comment (maximumpower55) [reply src/main/java/net/coderbot/iris/mixin/MixinScreenEffectRenderer.java]: Fixed

**Review by coderbot16 — APPROVED** [review]
Thanks!

**Pull request merged**